### ISSUE TITLE
Switch expression syntax results in false positive of switch_case_alignment

### ISSUE THREAD

**Issue by robertmryan** [realm/SwiftLint#5227]
### New Issue Checklist

- [✓] Updated SwiftLint to the latest version
- [✓] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

In new `switch` expression syntax in Swift 5.9 (as discussed in [SE-0380](https://github.com/apple/swift-evolution/blob/main/proposals/0380-if-switch-expressions.md)) can result in false positive `switch_case_alignment`:

```swift
class Experiment {
    enum ExperimentalEnum {
        case foo, bar
    }

    func exampleWorkingFine(value: ExperimentalEnum) -> Bool {
        print(#function)

        switch value {
        case .foo: return true
        case .bar: return false
        }
    }

    func exampleWithFalsePositive(value: ExperimentalEnum) -> Bool {
        print(#function)

        return switch value {
        case .foo: true  // Switch and Case Statement Alignment Violation: Case statements should vertically aligned with their enclosing switch statement (switch_case_alignment)
        case .bar: false // Switch and Case Statement Alignment Violation: Case statements should vertically aligned with their enclosing switch statement (switch_case_alignment)
        }
    }

    func exampleWithWorkAround(value: ExperimentalEnum) -> Bool {
        print(#function)

        return switch value {
               case .foo: true
               case .bar: false
               }
    }
}
```

E.g.:

<img width="1565" alt="CleanShot 2023-09-18 at 17 13 53@2x" src="https://github.com/realm/SwiftLint/assets/1530768/75a1e901-89ad-4262-9a73-b6655402df38">

It looks like it is looking for alignment with the `s` in `switch`, and it arguably should be lining up with the first character of that statement. I guess the `exampleWithWorkAround` isn’t too bad, but that’s not where Xcode will put it (and it will get reset if you choose “Editor” » “Structure” » “Re-Indent”). It seems like a bug if the indentation enforces a rule other than what the IDE uses.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Linting Swift files in current working directory
Linting 'ViewController.swift' (1/3)
Linting 'Experiment.swift' (2/3)
Linting 'AppDelegate.swift' (3/3)
/Users/robertryan/Library/CloudStorage/Dropbox/Private/Xcode/Temp/SwiftLintBug/SwiftLintBug/Experiment.swift:28:9: warning: Switch and Case Statement Alignment Violation: Case statements should vertically aligned with their enclosing switch statement (switch_case_alignment)
/Users/robertryan/Library/CloudStorage/Dropbox/Private/Xcode/Temp/SwiftLintBug/SwiftLintBug/Experiment.swift:29:9: warning: Switch and Case Statement Alignment Violation: Case statements should vertically aligned with their enclosing switch statement (switch_case_alignment)
Done linting! Found 2 violations, 0 serious in 3 files.
```

### Environment

* SwiftLint version: 0.52.4
* Installation method used: CocoaPods
* Paste your configuration file: None used

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)?

    > Xcode 15.0
    > Build version 15A240d

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```none
$ ./Pods/SwiftLint/swiftlint lint --no-cache --enable-all-rules SwiftLintBug/Experiment.swift
Linting Swift files at paths SwiftLintBug/Experiment.swift
Linting 'Experiment.swift' (1/1)
warning: The `anyobject_protocol` rule is now deprecated and will be completely removed in a future release.
warning: The `inert_defer` rule is now deprecated and will be completely removed in a future release.
warning: The `unused_capture_list` rule is now deprecated and will be completely removed in a future release.
/…/SwiftLintBug/SwiftLintBug/Experiment.swift:11:5: warning: Explicit ACL Violation: All declarations should specify Access Control Level keywords explicitly (explicit_acl)
/…/SwiftLintBug/SwiftLintBug/Experiment.swift:15:5: warning: Explicit ACL Violation: All declarations should specify Access Control Level keywords explicitly (explicit_acl)
/…/SwiftLintBug/SwiftLintBug/Experiment.swift:24:5: warning: Explicit ACL Violation: All declarations should specify Access Control Level keywords explicitly (explicit_acl)
/…/Temp/SwiftLintBug/SwiftLintBug/Experiment.swift:10:1: warning: Explicit ACL Violation: All declarations should specify Access Control Level keywords explicitly (explicit_acl)
/…/Temp/SwiftLintBug/SwiftLintBug/Experiment.swift:10:1: warning: Explicit Top Level ACL Violation: Top-level declarations should specify Access Control Level keywords explicitly (explicit_top_level_acl)
/…/Temp/SwiftLintBug/SwiftLintBug/Experiment.swift:10:1: warning: Required Deinit Violation: Classes should have an explicit deinit method (required_deinit)
/…/SwiftLintBug/SwiftLintBug/Experiment.swift:28:9: warning: Switch and Case Statement Alignment Violation: Case statements should vertically aligned with their enclosing switch statement (switch_case_alignment)
/…/SwiftLintBug/SwiftLintBug/Experiment.swift:29:9: warning: Switch and Case Statement Alignment Violation: Case statements should vertically aligned with their enclosing switch statement (switch_case_alignment)
/…/SwiftLintBug/SwiftLintBug/Experiment.swift:19:9: warning: Cases on Newline Violation: Cases inside a switch should always be on a newline (switch_case_on_newline)
/…/SwiftLintBug/SwiftLintBug/Experiment.swift:20:9: warning: Cases on Newline Violation: Cases inside a switch should always be on a newline (switch_case_on_newline)
/…/SwiftLintBug/SwiftLintBug/Experiment.swift:28:9: warning: Cases on Newline Violation: Cases inside a switch should always be on a newline (switch_case_on_newline)
/…/SwiftLintBug/SwiftLintBug/Experiment.swift:29:9: warning: Cases on Newline Violation: Cases inside a switch should always be on a newline (switch_case_on_newline)
Done linting! Found 12 violations, 0 serious in 1 file.
```

**Comment by robertmryan** [comment]
Maybe the rule should be how `case` lines up with the closing brace (allowing the user to apply whatever indentation they want after the `return`).

**Comment by SimplyDanny** [comment]
I guess that's "fixed" by #5193 in that switch expressions are completely ignored from the rule. That provides some time to think about a proper solution.

> Maybe the rule should be how `case` lines up with the closing brace (allowing the user to apply whatever indentation they want after the `return`).

I like this idea.

**Comment by SimplyDanny** [comment]
Fixed by #5229.
